libdnf5-plugins/actions/actions.cpp

```diff
@@ -844,7 +854,777 @@ void Actions::process_command_output_line(std::string_view line) {
+                        {"vendor",
+                         [](const base::TransactionPackage *, const rpm::Package & pkg) { return pkg.get_vendor(); }}};
+
+                unsigned int requested_attrs = 0;
+                auto * joutput = get_array(jargs, "output");
+                for (size_t idx = 0; idx < json_object_array_length(joutput); ++idx) {
+                    const auto requested_attr = get_string_view(json_object_array_get_idx(joutput, idx));
+                    for (std::size_t idx = 0; idx < sizeof(attrs_list) / sizeof(attrs_list[0]); ++idx) {
+                        if (attrs_list[idx].first == requested_attr) {
+                            requested_attrs |= 1 << idx;
+                        }
+                    }
+                }
+
+                libdnf5::sack::ExcludeFlags query_flags{0};
+                auto * jparams = get_any_object_or_null(jargs, "params");
+                if (jparams) {
+                    get_array(jargs, "params");  // check, must be array
+                    for (size_t idx = 0; idx < json_object_array_length(jparams); ++idx) {
+                        auto * jparam = json_object_array_get_idx(jparams, idx);
+                        const auto param_key = get_string_view(jparam, "key");
+                        //auto * jvalue = get_any_object_or_null(jparam, "value");
+                        //const auto value = jvalue ? std::string(get_string_view(jvalue)) : "";
+                        if (param_key == "IGNORE_EXCLUDES") {
+                            query_flags = query_flags | libdnf5::sack::ExcludeFlags::IGNORE_EXCLUDES;
+                        } else if (param_key == "IGNORE_MODULAR_EXCLUDES") {
+                            query_flags = query_flags | libdnf5::sack::ExcludeFlags::IGNORE_MODULAR_EXCLUDES;
+                        } else if (param_key == "IGNORE_REGULAR_EXCLUDES") {
+                            query_flags = query_flags | libdnf5::sack::ExcludeFlags::IGNORE_REGULAR_EXCLUDES;
+                        } else if (param_key == "IGNORE_REGULAR_CONFIG_EXCLUDES") {
+                            query_flags = query_flags | libdnf5::sack::ExcludeFlags::IGNORE_REGULAR_CONFIG_EXCLUDES;
+                        } else if (param_key == "IGNORE_REGULAR_USER_EXCLUDES") {
+                            query_flags = query_flags | libdnf5::sack::ExcludeFlags::IGNORE_REGULAR_USER_EXCLUDES;
+                        } else {
+                            throw JsonRequestError(fmt::format("Bad key \"{}\" for params", param_key));
+                        }
+                    }
+                }
+
+                auto * jfilters = get_any_object_or_null(jargs, "filters");
+                if (jfilters) {
+                    get_array(jargs, "filters");  // check, must be array
+                }
+
+                enum class Direction { IN, OUT, ALL } edirection = Direction::ALL;
```

Comment:
Why isn't the `Action::Direction` used here?

Reply:
Thank you. Good point. This is a good candidate for using the `Action::Direction` type. I somehow missed it when I reorganized the code.